### ISSUE TITLE
Discussion: Optional use of ngrok/tunneling software to connect to a remote bot

### ISSUE THREAD

**Issue by cwhitten** [Microsoft/BotFramework-Emulator#1307]
Currently the Emulator requires [ngrok](https://ngrok.com/) to connect to remotely hosted bots. We have received feedback and requests to not require this tool, or any additional tunneling software, giving the users the option to reconfigure firewalls and expose any necessary ports to facilitate the network communication that the Emulator needs with the remote bot.

There are a few ways to achieve this, and we would like to hear from stakeholders any details around requiring ngrok and the friction it causes different bot developers and any expectations they have in terms of UX if we were to introduce tunneling software as optional. Please use this thread to discuss with the Bot Framework team.

**Comment by justinwilaby** [comment]
ngrok is not a unique tool and there are many ways to accomplish what it provides. I would support the complete removal of ngrok from the Emulator to allow the developer greater flexibility when exposing a locally running service. 

For those who may rely on ngrok currently, I would propose directing them to the ngrok documents or provide a dedicated entry in our hosted Emulator docs.

**Comment by ryanvolum** [comment]
I'm inclined to agree with @justinwilaby here - there are several different ways to accomplish tunneling (I even think the Service Bus team has one), so locking developers into ngrok (where they don't know where their messages are being bounced off) feels heavy-handed.

**Comment by tonyanziano** [comment]
I agree and believe that network configuration should be something handled by the user. I don't think that we can reliably provide a "one size fits all" solution for all the custom network configurations out there.
